Add minutes from 21 Mar 2022 Panel meeting (#211)
solid/authentication-panel: meetings/2022-03-21.md

> none
> The Client does not authenticate itself at the Token Endpoint, either because it uses only the Implicit Flow (and so does not use the Token Endpoint) or because it is a Public Client with no Client Secret or other authentication mechanism.

* eP: we should clarify what confidential clients do
* AC: are we going to restate what OIDC or OAuth2 already says about it?
* eP: How do we see this working in practice? In some implementations, it seems to work with `none`?
* AvdB: the spec mentions client registration. Where would the `none` be set?
* AvdB: when you register a client (e.g., Dynamic Client Registration), you can set a particular method. For Client Identifiers, these are public clients
* eP: I wasn't sure if this is for the token endpoint
* AC: Nature of client identfires is that these are public clients. If it is confidential clients, the authentication method is important. Those client identifiers are intended to be public clients. Maybe we should be clear about it the spec.
* eP: for use with confidential clients, in interop panel, the authorization agent acts as a confidential client

> * AC: what we define is about Public Clients but we don't say anything about confidential clients. If impl supports confidential clients it can do anything it likes provided that it follows OAuth 2.0.

Reviewer: ```suggestion
* AC: what we define is about Public Clients, but we don't say anything about confidential clients. If impl supports confidential clients, it can do anything it likes, provided that it follows OAuth 2.0.
```

* eP: can Public Client get a refresh token?
* AC: There needs to be interaction with end-user to confirm refresh token.
* AC: I'm not a fan of refresh tokens in the browser, but given browser limitations around iframes, they are useful in certain cases.

> * eP: it's recommended that an ID token is short-lived, so how do you get a token that isn't expired
> * AC: One approach is using refresh tokens, another approach is rather than redirecting user away, open new window and direct user through the auth flow. Given that the have cookie there should not be any interaction

Reviewer: ```suggestion
* eP: it's recommended that an ID token is short-lived, so how do you get a token that isn't expired?
* AC: One approach is using refresh tokens. Another approach is, rather than redirecting user away, open new window, and direct user through the auth flow. Given that they have cookie, there should not be any interaction.
```

* eP: the prompt=none
* eP: for now, we can make this an in-line issue to make sure there's no problem/constraints with public clients
* eP: likely not before FPWD
* eP: for server-side client impl, we plan to use refresh tokens. With client credentials, it might be more attractive to use those

### Presenting multiple WebIDs in an ID Token

[Issue#47](https://github.com/solid/solid-oidc/issues/47)

* AvdB: This may have been related to a blogpost about a WebID being a 1-1 relationship with a User's identity. E.g., one person can have multiple WebIDs, we may need the ability to put multiple WebIDs in a token.
* eP: Correlating WebIDs seems against the point of the blog post, because one may want to keep those identities separate, from a privacy perspective
* eP: If an OP puts all identities in a single token, that seems like a problem for privacy

> * AC: I have multiple identities for different contexts, personal, work, open source etc. There are different trust domains and I like to manage them distinctly. When I login through some corporate id I don't get access to something I have access to outside work. And vice versa.

Reviewer: ```suggestion
* AC: I have multiple identities for different contexts, personal, work, open source, etc. There are different trust domains and I like to manage them distinctly. When I log in through some corporate ID, I don't get access to something I have access to outside work. And vice versa.
```

> * ...: Sometimes you want to say that a person has different identifiers and they should be able to use any of them. We can see nowadays people logging with google, twitter, facebook etc.
> * ...: I would argue that ID Token is scoped to a particular WebID. If it manages this identity it should focus on that. We could focus on using multiple ID Tokens and possibly exchanging them at UMA AS. It can be more managable if we keep them separate from the start. For example building VCs based on separate identities. For ID Token there is focus on single WebID.

Reviewer: ```suggestion
* ...: Sometimes you want to say that a person has different identifiers and they should be able to use any of them. We can see nowadays people logging in with Google, Twitter, Facebook, etc.
* ...: I would argue that an ID Token is scoped to a particular WebID. If it manages this identity, it should focus on that. We could focus on using multiple ID Tokens and possibly exchanging them at UMA AS. It can be more manageable if we keep them separate from the start. For example, building VCs based on separate identities. For ID Token, there is focus on single WebID.
```

* AvdB: email is probably a bad analogy, since there is a separate login. If all the WebIDs are in the same place, then it would also see all my other WebIDs at a given OP.
* eP: any objections to making this an inline issue? Perhaps mention in primer about using multiple ID tokens
* eP: "The webid claim must be the user's WebID"
* AvdB: I'm also not enitrely certain about this use case. It needs more clarification

### Accessing NonRDF-Sources directly via browser

[Issue#31](https://github.com/solid/solid-oidc/issues/31)

* eP: Created an issue in the specification repository
* ...: Ruben mentioned the "Databrowser Hack"
* ...: Maybe we can just close this issue in Solid-OIDC and continue in the Solid Protocol spec issue
* AC: +1

ACTION: eP: close this issue and track the discussion in the protocol specification

### Overview how client credential is coming along

> * AvdB: in various meetings there have been some discussions about how client credentials would work in the Solid Ecosystem. What is the status?

Reviewer: ```suggestion
* AvdB: in various meetings, there have been some discussions about how client credentials would work in the Solid Ecosystem. What is the status?
```

* eP: https://github.com/solid/solid-oidc/blob/main/alternative-flows.md
* eP: I think we should center the discussion around the document referenced here.
* ...: We should start creating issues where these items can be discussed

> * AC: This issue has been around for quite some time. There are a number of experiments. There is a larger question about how we should specify that. Sometimes you want some exprimentation before specifying a particular pattern. I would be interested in client credentials in a more comprehensive way. What are other flows that we have in Solid. See if there are shared patterns.

Reviewer: ```suggestion
* AC: This issue has been around for quite some time. There are a number of experiments. There is a larger question about how we should specify that. Sometimes you want some experimentation before specifying a particular pattern. I would be interested in client credentials in a more comprehensive way. What are other flows that we have in Solid? See if there are shared patterns.
```

* ...: Rather than have series of false starts, we should get a good sense of what we would like to try to do first.
* eP: I likely won't have time to focus on this during March, but hope to pick this up in April and later, esp with public key authentication. Consent is another, related area.

> * AC: The question of public/private key is pretty powerful. In a way it gets us back to WebID-TLS but we could avoid UX issues which it had.
> * AC: Credentials Community group has been working on various solutions. CHAPI, DID-Comm etc. We've been really focused on OIDC but we should complete it and strat looking at other interesting approches which can be more flexible.

Reviewer: ```suggestion
* AC: The question of public/private key is pretty powerful. In a way it gets us back to WebID-TLS, but we could avoid UX issues which it had.
* AC: Credentials Community group has been working on various solutions. CHAPI, DID-Comm, etc. We've been really focused on OIDC, but we should complete it and start looking at other interesting approaches which can be more flexible.
```

* eP: on the topic of public/private key, Henry has been using this with HTTP signatures. DPoP and GNAP also use proof of possession mechanisms. The common part can be about the publication of public keys.
* ...: it would be good to clarify how these public keys can be published in Solid, one of the future focus areas after Solid-OIDC
* AC: We wondered a little bit away. We should answer some of the questions brought up.
* AvdB: In general, we have been looking into and documenting how client credentials can be used with Solid, but we will want to explore some of the other flows, too

> * eP: for me, the key question is that some of the OIDC test suites might fail if client credentials is supported. There is still the question of consent and whether a user approves that the client can use the ID token. This consent mechanism may span these different flows.

Reviewer: ```suggestion
* eP: for me, the key question is whether some of the OIDC test suites might fail if client credentials are supported. There is still the question of consent, and whether a user approves that the client can use the ID token. This consent mechanism may span these different flows.
```
